Add routing doc (#1835)
spryker/spryker-docs: _drafts/oryx/routing.md

There are also framework specific integrations for providing actual routing configuration:

- for Lit use `@spryker-oryx/router/lit` package

## Getting started

To start using router add `RouterFeature` to your Oryx application:

```ts
import { appBuilder } from '@spryker-oryx/core';
import { RouterFeature } from '@spryker-oryx/router';

appBuilder().withFeature(new RouterFeature());
```

Now you can use `RouterService` in your components and services to interact with the router.

## Lit routing

To render router outlet you should add `LitRouter` to your root app component and render it's outlet:

```ts
import { LitElement } from 'lit';
import { LitRouter } from '@spryker-oryx/router/lit';

class RootComponent extends LitElement {
  router = new LitRouter(this, []); // <-- You may provide some static routes here

  protected override render() {
    return this.router.outlet(); // <-- Render router outlet here
  }
}
```

Then you can provide routes for Lit applications using DI:

```ts
import { appBuilder } from '@spryker-oryx/core';
import { provideLitRoutes } from '@spryker-oryx/router/lit';

> const routes = [{ path: '/', render: () => html`<my-page></my-page>` }];

Reviewer: maybe we can second route example with parameter here?

# Routing

Routing is an essential feature for any web application, allowing users to navigate between different pages or components within the application. The `@spryker-oryx/router` package

> provides routing infrastructure for your application, including a `RouterService` for interacting with the router and framework-specific integrations for providing actual routing config.

Reviewer 2: Please split the last sentence into two. The last part, in particular, doesn't make sense for me

This guide will walk you through the process of setting up routing in your application, including how to add the `RouterFeature` to your application, render the router outlet, and provide routes using Dependency Injection (DI).

> `@spryker-oryx/router` provides infrastructure for routing capabilities. It gives you mainly the `RouterService` which is a generic way for navigating in your application.

Reviewer 2: It's the same info as in the first paragraph. Where would it make more sense?

Author: I guess we can remove the second sentence from the last paragraph to reduce duplicated info.

Author: I've rearranged it and removed it from the first paragraph.

import { RouterFeature } from '@spryker-oryx/router';

appBuilder().withFeature(new RouterFeature());
```

Now you can use `RouterService` in your components and services to interact with the router.

## Lit routing

To render router outlet you should add `LitRouter` to your root app component and render it's outlet:

```ts
import { LitElement } from 'lit';
import { LitRouter } from '@spryker-oryx/router/lit';

class RootComponent extends LitElement {
  router = new LitRouter(this, []); // <-- You may provide some static routes here

  protected override render() {
    return this.router.outlet(); // <-- Render router outlet here
  }
}
```

Then you can provide routes for Lit applications using DI:

```ts
import { appBuilder } from '@spryker-oryx/core';
import { provideLitRoutes } from '@spryker-oryx/router/lit';

const routes = [
  { path: '/', render: () => html`<my-page></my-page>` },
  { path: '/example/:id', render: ({ id }) => html`<my-example-page id=${id}></my-example-page>` },
];

appBuilder().withProviders(provideLitRoutes({ routes }));
```

Please note that this is a basic example and you may need to adjust the routing config
to fit your specific use case.

> Lit routes using standard [lit router library](https://github.com/lit/lit/tree/main/packages/labs/router#readme) so you can read it's documentation for more information on how to configure routes.

Author: There is already a link to docs https://github.com/lit/lit/tree/main/packages/labs/router#readme

---
title: Routing
description:
template: concept-topic-template

> last_updated: Apr 4, 2023

Reviewer 3: ```suggestion
last_updated: May 25, 2023
```

`@spryker-oryx/router` provides the infrastructure for routing capabilities. Its main component is `RouterService`, which enables navigation from one view to the next as users perform application tasks.

There are also framework-specific integrations for providing an actual routing configuration: for Lit, use the `@spryker-oryx/router/lit` package.

## Get started with routing

To start using the router, add `RouterFeature` to your Oryx application:

```ts
import { appBuilder } from '@spryker-oryx/core';
import { RouterFeature } from '@spryker-oryx/router';

appBuilder().withFeature(new RouterFeature());
```

Now you can use `RouterService` in your components and services to interact with the router.

## Lit routing

To render a router outlet, add `LitRouter` to your root app component and render its outlet:

```ts
import { LitElement } from 'lit';
import { LitRouter } from '@spryker-oryx/router/lit';

class RootComponent extends LitElement {
  router = new LitRouter(this, []); // <-- You can provide some static routes here

  protected override render() {
    return this.router.outlet(); // <-- Render router outlet here
  }
}
```

> Then, you can provide routes for Lit applications using DI:

Reviewer 3: Is this step optional?

Reviewer 2: This is more of a concept section than a guide

```ts
import { appBuilder } from '@spryker-oryx/core';
import { provideLitRoutes } from '@spryker-oryx/router/lit';

const routes = [
  { path: '/', render: () => html`<my-page></my-page>` },
  { path: '/example/:id', render: ({ id }) => html`<my-example-page id=${id}></my-example-page>` },
];

appBuilder().withProviders(provideLitRoutes({ routes }));
```
{% info_block infoBox %}

This is a basic example, and you may need to adjust the routing config to fit your specific use case.

{% endinfo_block %}

> Lit routes using the standard [lit router library](https://github.com/lit/lit/tree/main/packages/labs/router#readme), so see its documentation for more information on how to configure routes.

Reviewer 3: I would suggest adding a link to the official documentation.
```suggestion
Lit routes use the standard [lit router library](https://github.com/lit/lit/tree/main/packages/labs/router#readme). For more information about configuring routes, see its documentation.
```
